common/dictionary-db/dictionary-provider.ts

```diff
@@ -0,0 +1,85 @@
+import {
+    DictionaryLocalTokenInput,
+    DictionaryTokenKey,
+    LemmaResults,
+    TokenResults,
+} from '@project/common/dictionary-db';
+import { DictionaryBuildAnkiCacheState } from '@project/common';
+import { AsbplayerSettings } from '@project/common/settings';
+
+export interface DictionaryStorage {
+    getBulk: (profile: string | undefined, track: number, tokens: string[]) => Promise<TokenResults>;
+    getByLemmaBulk: (profile: string | undefined, track: number, lemmas: string[]) => Promise<LemmaResults>;
+    saveRecordLocalBulk: (
+        profile: string | undefined,
+        localTokenInputs: DictionaryLocalTokenInput[]
+    ) => Promise<DictionaryTokenKey[]>;
+    deleteRecordLocalBulk: (profile: string | undefined, tokens: string[]) => Promise<number>;
+    deleteProfile: (profile: string) => Promise<[number, number, number]>;
+    buildAnkiCache: (
+        profile: string | undefined,
+        settings: AsbplayerSettings,
+        options?: { useOriginTab?: boolean }
+    ) => Promise<DictionaryBuildAnkiCacheState>;
+    addBuildAnkiCacheStateChangeCallback: (callback: (message: DictionaryBuildAnkiCacheState) => void) => void;
+    removeBuildAnkiCacheStateChangeCallback: (callback: (message: DictionaryBuildAnkiCacheState) => void) => void;
+    ankiCardWasModified: () => void;
+    addAnkiCardModifiedCallback: (callback: () => void) => void;
+    removeAnkiCardModifiedCallback: (callback: () => void) => void;
```

Comment:
It's probably fine in your diff but this type of API design for event unsubscription scares me because it's easy to forget to call `removeAnkiCardModifiedCallback` or accidentally call `addAnkiCardModifiedCallback` many times in a row. My personal preference is to return an unsubscription callback from every subscription method.